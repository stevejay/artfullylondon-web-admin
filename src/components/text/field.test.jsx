import React from 'react'
import { shallow } from 'enzyme'
import _ from 'lodash'

import TextField from '_src/components/text/field'
import Text from '_src/components/text'

it('should render correctly', () => {
  const wrapper = shallow(
    <TextField
      label='The Label'
      tooltip='The Tooltip'
      input={{ value: 'term', onChange: _.noop }}
      meta={{ touched: false, error: null }}
      maxLength={50}
      required={false}
      disabled={false}
      autoFocus={false}
      placeholder='The Placeholder'
      password={false}
      autos={false}
      forceSingleLine={false}
      remainingChars={false}
    />
  )

  expect(wrapper).toMatchSnapshot()
})

it('should render correctly when has remaining chars', () => {
  const wrapper = shallow(
    <TextField
      label='The Label'
      tooltip='The Tooltip'
      input={{ value: 'term', onChange: _.noop }}
      meta={{ touched: false, error: null }}
      maxLength={50}
      required={false}
      disabled={false}
      autoFocus={false}
      placeholder='The Placeholder'
      password={false}
      autos={false}
      forceSingleLine={false}
      remainingChars
    />
  )

  expect(wrapper).toMatchSnapshot()
})

it('should render correctly with an error', () => {
  const wrapper = shallow(
    <TextField
      label='The Label'
      tooltip='The Tooltip'
      input={{ value: 'term', onChange: _.noop }}
      meta={{ touched: true, error: 'The Error' }}
      maxLength={50}
      required={false}
      disabled={false}
      autoFocus={false}
      placeholder='The Placeholder'
      password={false}
      autos={false}
      forceSingleLine={false}
      remainingChars={false}
    />
  )

  expect(wrapper).toMatchSnapshot()
})

it('should handle the text input control getting and losing focus', () => {
  const wrapper = shallow(
    <TextField
      label='The Label'
      tooltip='The Tooltip'
      input={{ value: 'term', onChange: _.noop }}
      meta={{ touched: false, error: null }}
      maxLength={50}
      required={false}
      disabled={false}
      autoFocus={false}
      placeholder='The Placeholder'
      password={false}
      autos={false}
      forceSingleLine={false}
      remainingChars={false}
    />
  )

  expect(wrapper.state()).toEqual({ active: false })

  wrapper.find(Text).simulate('focus')
  wrapper.update()

  expect(wrapper.state()).toEqual({ active: true })

  wrapper.find(Text).simulate('blur')
  wrapper.update()

  expect(wrapper.state()).toEqual({ active: false })
})

describe('shouldComponentUpdate', () => {
  it('should not update when props have not changed', () => {
    const wrapper = shallow(
      <TextField
        label='The Label'
        tooltip='The Tooltip'
        input={{ value: 'term', onChange: _.noop }}
        meta={{ touched: false, error: null }}
        maxLength={50}
        required={false}
        disabled={false}
        autoFocus={false}
        placeholder='The Placeholder'
        password={false}
        autos={false}
        forceSingleLine={false}
        remainingChars={false}
      />
    )

    const result = wrapper.instance().shouldComponentUpdate(
      {
        input: { value: 'term' },
        meta: { touched: false, error: null },
        disabled: false
      },
      { active: false }
    )

    expect(result).toEqual(false)
  })

  it('should update when props have changed', () => {
    const wrapper = shallow(
      <TextField
        label='The Label'
        tooltip='The Tooltip'
        input={{ value: 'term', onChange: _.noop }}
        meta={{ touched: false, error: null }}
        maxLength={50}
        required={false}
        disabled={false}
        autoFocus={false}
        placeholder='The Placeholder'
        password={false}
        autos={false}
        forceSingleLine={false}
        remainingChars={false}
      />
    )

    const result = wrapper.instance().shouldComponentUpdate(
      {
        input: { value: 'term' },
        meta: { touched: false, error: null },
        disabled: false
      },
      { active: true }
    )

    expect(result).toEqual(true)
  })
})
