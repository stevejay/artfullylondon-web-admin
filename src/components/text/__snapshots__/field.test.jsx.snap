// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Container
  disabled={false}
  error={null}
  htmlFor="The Label"
  label="The Label"
  required={false}
  tooltip="The Tooltip"
  touched={false}
>
  <Text
    autoFocus={false}
    autos={false}
    disabled={false}
    error={null}
    forceSingleLine={false}
    id="The Label"
    maxLength={50}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    password={false}
    placeholder="The Placeholder"
    readOnly={false}
    touched={false}
    value="term"
  />
</Container>
`;

exports[`should render correctly when has remaining chars 1`] = `
<Container
  disabled={false}
  error={null}
  htmlFor="The Label"
  label="The Label"
  required={false}
  tooltip="The Tooltip"
  touched={false}
>
  <RemainingChars
    active={false}
    maxLength={50}
    value="term"
  />
  <Text
    autoFocus={false}
    autos={false}
    disabled={false}
    error={null}
    forceSingleLine={false}
    id="The Label"
    maxLength={50}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    password={false}
    placeholder="The Placeholder"
    readOnly={false}
    touched={false}
    value="term"
  />
</Container>
`;

exports[`should render correctly with an error 1`] = `
<Container
  disabled={false}
  error="The Error"
  htmlFor="The Label"
  label="The Label"
  required={false}
  tooltip="The Tooltip"
  touched={true}
>
  <Text
    autoFocus={false}
    autos={false}
    disabled={false}
    error="The Error"
    forceSingleLine={false}
    id="The Label"
    maxLength={50}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    password={false}
    placeholder="The Placeholder"
    readOnly={false}
    touched={true}
    value="term"
  />
</Container>
`;
